### DISCUSSION TITLE
smtp config

### DISCUSSION THREAD

**Discussion by andyzhangwp** [supabase/supabase#27434] [Questions] [answered]
How can I configure my own email provider in config.toml to replace Inbucket in sending emails, just like configuring Inbucket?

**Comment by GaryAustin1** [comment, marked as answer]
If you mean for local development with the CLI (different from self hosting) there is no support for SMTP.
https://github.com/orgs/supabase/discussions/22677

**Reply by andyzhangwp** [reply]
After configuring STMP, I encountered the following issue during Supabase start：
Unknown config fields: [auth.email.smtp_host auth.email.smtp_port auth.email.smtp_user auth.email.smtp_pass auth.email.smtp_sender_name auth.email.smtp_sender_email]


config.toml ：
......
[inbucket]
enabled = false

[auth.email]
enable_signup = true
double_confirm_changes = true
enable_confirmations = true

# SMTP settings for sending emails
smtp_host = "your.smtp.host"
smtp_port = 587
smtp_user = "your-smtp-username"
smtp_pass = "your-smtp-password"
smtp_sender_name = "Your Sender Name"
smtp_sender_email = "your-sender-email@example.com"

**Reply by GaryAustin1** [reply]
As I understand it (and the link mentions) you can't do SMTP with local dev CLI.

**Reply by andyzhangwp** [reply]
I understand, thank you!  @GaryAustin1

**Comment by KristiyanBozhankovFP** [comment]
I was missing the config.toml file and with `supabase init` I was able to generate it.

Then all you need to do is update these settings and restart the supabase containers with `supabase stop` & `supabase start`:
```
# Use a production-ready SMTP server
[auth.email.smtp]
enabled = true
host = "env(SMTP_HOST)"
port = 465
user = "env(SMTP_USER)"
pass = "env(SMTP_PASS)"
admin_email = "admin@admin.com"
sender_name = "Admin"
```
